src/Ferrite.jl

```diff
@@ -23,11 +24,13 @@ struct RefCube <: AbstractRefShape end
 struct RefPrism <: AbstractRefShape end
 
 """
-Abstract type which has `CellValues` and `FaceValues` as subtypes
+Abstract type which has `CellValues` and `FaceValues` as subtypes. Each Value type is
+equipped with the spatial dimension `sdim` of the grid, the reference elements' dimension
+`rdim`, the data type for the values `T` and the reference element's geometry `refshape`.
 """
-abstract type Values{dim,T,refshape} end
-abstract type CellValues{dim,T,refshape} <: Values{dim,T,refshape} end
-abstract type FaceValues{dim,T,refshape} <: Values{dim,T,refshape} end
+abstract type Values{sdim,rdim,T,refshape} end
+abstract type CellValues{sdim,rdim,T,refshape} <: Values{sdim,rdim,T,refshape} end
+abstract type FaceValues{sdim,rdim,T,refshape} <: Values{sdim,rdim,T,refshape} end
```

Comment:
We should think about which of the parameters make sense to have on the abstract types here, it would be good to only keep the ones that will ever be used by dispatch.

Reply:
I think from the user side of the interface it makes sense to have access to these parameters to control dispatch on a higher level (e.g. one dispatch for assembly on triangles).

Reply:
Maybe a different angle then. What would a user likely use to dispatch on?

Reply:
I have never really used the parameters. Just like in the examples, I only add `::CellValues` to not pass the wrong argument at the wrong slot.

I suppose that it could sometimes be useful to have dim to make sure multiple values have the same ome (`::CellValues{dim}, ::FaceValues{dim}`) but usually you dont have multiple dims in flight in the same script anyway.